dipy/tracking/local/optimizedlocaltracking.py

```diff
@@ -0,0 +1,420 @@
+
+from pyspark import  SparkContext
+# Note: you can only run this code once before getting an error if you don't run sc.stop()
+sc = SparkContext( 'local', 'pyspark')
+"""
+
+#nums = sc.parallelize(xrange(1000))
+#def duh(x):
+#    sdfjkalsdhaf
+#new_nums = nums.map(duh)
+#print str(new_nums.collect())
+
+# enum TissueClass (tissue_classifier.pxd) is not accessible
+# from here. To be changed when minimal cython version > 0.21.
+# cython 0.21 - cpdef enum to export values into Python-level namespace
+# https://github.com/cython/cython/commit/50133b5a91eea348eddaaad22a606a7fa1c7c457
+TissueTypes = Bunch(OUTSIDEIMAGE=-1, INVALIDPOINT=0, TRACKPOINT=1, ENDPOINT=2)
+
+#N = self.maxlen
+#dg = self.direction_getter
+#tc = self.tissue_classifier
+#ss = self.step_size
+#fixed = self.fixed
+#max_cross = self.max_cross
+#vs = self._voxel_size
+#
+#inv_A = np.linalg.inv(self.affine)
+#lin = inv_A[:3, :3]
+#offset = inv_A[:3, 3]
+#
+#F = np.empty((N + 1, 3), dtype=float)
+#B = F.copy()
+
+#SEEDS = 0
+
+
+# run normally
+# see if specific arguments can be "pickled" or passed into Pool
+
+
+
+def streamline_computation_wrapper(streamline_arguments):
+    # print "got in wrapper class: "
+    # print len(streamline_arguments.seed)
+    # print len(streamline_arguments.config_args)
+    # print
+    return streamline_computation(streamline_arguments.seed, *streamline_arguments.config_args)
+
+class StreamlineArguments():
+    """
+    Wrapper class to hold streamline arguments
+    """
+    def __init__(self, seed, config_args):
+        self.seed = seed
+        self.config_args = config_args
+
+
+def streamline_computation(s, inv_A, lin, offset, F, B, vs, dg, tc, affine,
+                 ss, max_cross=None, maxlen=500, fixed=True,
+                 return_all=True):
+    """
+    Helper function for parallelizing the computation of streamlines
+    """
+    # print "got in wrapper class: "    
+    s = np.dot(lin, s) + offset
+    directions = dg.initial_direction(s)
+    directions = directions[:max_cross]
+    for first_step in directions:
+        stepsF, tissue_class = local_tracker(dg, tc, s, first_step,
+                                             vs, F, ss, fixed)
+        if not (return_all or
+                tissue_class == TissueTypes.ENDPOINT or
+                tissue_class == TissueTypes.OUTSIDEIMAGE):
+            continue
+        first_step = -first_step
+        stepsB, tissue_class = local_tracker(dg, tc, s, first_step,
+                                             vs, B, ss, fixed)
+        if not (return_all or
+                tissue_class == TissueTypes.ENDPOINT or
+                tissue_class == TissueTypes.OUTSIDEIMAGE):
+            continue
+
+        if stepsB == 1:
+            streamline = F[:stepsF].copy()
+        else:
+            parts = (B[stepsB-1:0:-1], F[:stepsF])
+            streamline = np.concatenate(parts, axis=0)
+        return streamline
+
+class OptimizedLocalTracking(object):
```

Comment:
This class seems like it's mostly a copy of the `LocalTracking` class. You should either change the class directly or use inheritance. We're going to need you to do that before we merge it, plus it makes it much easier for me to see what you've changed :).